tianshou/data/batch.py

```diff
@@ -569,10 +569,12 @@ def empty_(self, index: Union[
             else:  # scalar value
                 warnings.warn('You are calling Batch.empty on a NumPy scalar, '
                               'which may cause undefined behaviors.')
-                if isinstance(v, (np.generic, Number)):
+                if isinstance(v, (np.number, Number)):
```

Comment:
Why not to add `np.bool_` here. `self.__dict__[k] *= 0` is doing the same as `self.__dict__[k] &= False`, so I don't see the point in creating a new if case.

Reply:
OK I just saw your comment. There is the same problem with native Python `bool` type. So maybe you were right in the firstt place, and using `self.__dict__[k] = v.__class__(0)` is the most robust approach after all.

Reply:
Agree. Although ``self.__dict__[k] = v.__class__(0)`` will lead to new memory allocation, the memory cost of scalar is negligible.

Reply:
Indeed.